xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -3860,6 +3860,8 @@ public void prepareResources(XWikiContext context)
     public XWikiUser checkAuth(XWikiContext context) throws XWikiException
     {
         return getAuthService().checkAuth(context);
+        UserAuthenticatedEvent newEvent = new UserAuthenticatedEvent();
+        getObservationManager().notify(context);
```

Comment:
You need to check the ObservationManager's API in order to understand how to use it and how to send events through it.

Also, make sure you use and IDE (Eclipse) when making this chances, as it helps you both to see the API and to avoid writing code that is not valid. Finally, build the module that you are modifying, in order to make sure your changes are valid and not breaking the build.